### PR TITLE
Support single machine scheduler policy in AML mode

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: docs/en_US/TrainingService/AMLMode.md [hunk 1]
```diff
@@ -81,6 +81,13 @@ amlConfig:
     * optional key, default 1. Used to specify the max concurrency trial number on a GPU device.
 * useActiveGpu
     * optional key, default false. Used to specify whether to use a GPU if there is another process. By default, NNI will use the GPU only if there is no other active process in the GPU.
+* machineScheduler
+    * optional key, default round-robin. Used to set the machine scheduler policy for trial job, values choose from `random, round-robin, single`.
+    ```
+    random: trial jobs will be scheduled into different nodes randomly.
+    round-robin: trial jobs will be scheduled into different node sequentially.
+    single: trial jobs will always be scheduled into same node.
```
Comment (QuanluZhang) [inline docs/en_US/TrainingService/AMLMode.md:89]: this description is a little confusing. this schedule always uses one node?

File: docs/en_US/TrainingService/AMLMode.md [hunk 2]
```diff
@@ -81,6 +81,13 @@ amlConfig:
     * optional key, default 1. Used to specify the max concurrency trial number on a GPU device.
 * useActiveGpu
     * optional key, default false. Used to specify whether to use a GPU if there is another process. By default, NNI will use the GPU only if there is no other active process in the GPU.
+* machineScheduler
```
Comment (QuanluZhang) [inline docs/en_US/TrainingService/AMLMode.md:84]: not suggest to expose this option in configuration. we can simply use the new scheduler for aml training service.

File: ts/nni_manager/common/manager.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ interface ExperimentParams {
     multiThread?: boolean;
     versionCheck?: boolean;
     logCollection?: string;
+    machineScheduler?: string;
```
Comment (QuanluZhang) [inline ts/nni_manager/common/manager.ts:29]: yes, we can put the scheduler here internally.

File: docs/en_US/TrainingService/AMLMode.md (hunk 2 above)
Comment (squirrelsc) [inline docs/en_US/TrainingService/AMLMode.md:84]: It shouldn't be a AML only feature, may be useful for other platforms.

File: docs/en_US/TrainingService/AMLMode.md (hunk 1 above)
Comment (squirrelsc) [inline docs/en_US/TrainingService/AMLMode.md:89]: It needs to explain what's value to user to schedule on same node.

File: ts/nni_manager/training_service/reusable/gpuScheduler.ts [hunk 4]
```diff
@@ -30,7 +30,7 @@ export class GpuScheduler {
 
     // private readonly machineExecutorMap: Set<TrialDetail>;
     private readonly log: Logger = getLogger();
-    private readonly policyName: SCHEDULE_POLICY_NAME = 'round-robin';
+    private policyName: SCHEDULE_POLICY_NAME = 'round-robin';
```
Comment (squirrelsc) [inline ts/nni_manager/training_service/reusable/gpuScheduler.ts:33]: low case name?

File: docs/en_US/TrainingService/AMLMode.md (hunk 2 above)
Comment (SparkSnail) [reply docs/en_US/TrainingService/AMLMode.md:84]: @QuanluZhang @squirrelsc  we need to discuss the scenario of this feature in the meeting, to set this single schuduler only for aml or for all of platforms. To expose the configuration or set it as internally.

File: ts/nni_manager/training_service/reusable/gpuScheduler.ts (hunk 4 above)
Comment (SparkSnail) [reply ts/nni_manager/training_service/reusable/gpuScheduler.ts:33]: policyname is not a valid word.

**Comment by SparkSnail** [comment]
close this pr, will refactor logic.

**Pull request closed without merging**